datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/LocalDateDeserializer.java

```diff
@@ -155,7 +155,8 @@ protected LocalDate _fromString(JsonParser p, DeserializationContext ctxt,
                 // JavaScript by default includes time in JSON serialized Dates (UTC/ISO instant format).
                 if (string.length() > 10 && string.charAt(10) == 'T') {
                    if (string.endsWith("Z")) {
-                       return LocalDateTime.ofInstant(Instant.parse(string), ZoneOffset.UTC).toLocalDate();
+                       return LocalDate.parse(string.substring(0, string.length() - 1),
+                               DateTimeFormatter.ISO_LOCAL_DATE_TIME);
```

Comment:
Seems to me that there is no need to go back and forth to zoned instant when simply trying to retain local value.
Tests all pass as well.